### PR TITLE
Split Row Swap

### PR DESCRIPTION
**Pull request by lueelu**
We exploited more fine-grained implementation during the Trailing Matrix Update step and the Row Swap step is splitted for overlapping a fraction of dgemm and row swap kernels. A new stream, row swap stream, is created to eliminate the some wait time of Row Swap by partitioning the trailing matrix into several NB columns (may be 3 or 4). This is feasible since Row Swap of chunk i+1 is independent of Trailing Matrix Update of chunk i. The following figure illustrates the case of overlap Update of chunk i with Row Swap of chunk i+1.

### CODE REVIEW

File: src/pgesv/HPL_pdupdateNN.cpp [hunk 1]
```diff
@@ -221,11 +221,10 @@ void HPL_pdupdateNN
       {
 #ifdef HPL_DETAILED_TIMING
          HPL_ptimer( HPL_TIMING_LASWP );
-         HPL_BE_dlaswp00N(jb, nn, Aptr, lda, ipiv, T_HIP);
+         // HPL_BE_dlaswp00N(jb, nn, Aptr, lda, ipiv, T_HIP);
```
Comment (reger-men) [inline src/pgesv/HPL_pdupdateNN.cpp]: Will this be eliminated?

File: src/pgesv/HPL_pdupdateNN.cpp [hunk 2]
```diff
@@ -221,11 +221,10 @@ void HPL_pdupdateNN
...
          HPL_ptimer( HPL_TIMING_LASWP );
 #else
-         HPL_BE_dlaswp00N(jb, nn, Aptr, lda, ipiv, T_HIP);
-
+         // HPL_BE_dlaswp00N(jb, nn, Aptr, lda, ipiv, T_HIP);
```
Comment (reger-men) [inline src/pgesv/HPL_pdupdateNN.cpp]: Same here?

File: src/pgesv/HPL_pdupdateNT.cpp [hunk 3]
```diff
@@ -230,10 +230,10 @@ void HPL_pdupdateNT
       {
 #ifdef HPL_DETAILED_TIMING
          HPL_ptimer( HPL_TIMING_LASWP );
-         HPL_BE_dlaswp00N( jb, nn, Aptr, lda, ipiv, T_HIP );
+         // HPL_BE_dlaswp00N( jb, nn, Aptr, lda, ipiv, T_HIP );
```
Comment (reger-men) [inline src/pgesv/HPL_pdupdateNT.cpp]: Same here.

File: src/pgesv/HPL_pdupdateNT.cpp [hunk 4]
```diff
@@ -230,10 +230,10 @@ void HPL_pdupdateNT
...
          HPL_ptimer( HPL_TIMING_LASWP );
 #else
-         HPL_BE_dlaswp00N( jb, nn, Aptr, lda, ipiv, T_HIP );
+         // HPL_BE_dlaswp00N( jb, nn, Aptr, lda, ipiv, T_HIP );
```
Comment (reger-men) [inline src/pgesv/HPL_pdupdateNT.cpp]: and here.

**Review by reger-men — COMMENTED** [review]
Please check the comments.

**Comment by lueelu** [comment]
Have eliminated these comments.

**Pull request merged**